### ISSUE TITLE
New Rule: An element with input type roles without an `aria-label` should be marked as invalid

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#1347]
# label-explicit

Ensures elements with explicit form roles have a label.

**Tags:** cat.forms, wcag2a, wcag332, wcag131, section508, section508.22.n

## Selector

`[role="textbox"],[role="checkbox"],[role="spinbutton"],[role="radio"],[role="slider"],[role="searchbox"]`

## Checks

Check that an element with a matching role has either an `aria-label` or `aria-labelledby`. Returns true if it does, fail otherwise.

**Comment by jdkahn** [comment]
I agree this should be fixed, but wondering if it makes more sense to add a new rule or modify the current one. The current rule only tests HTML `input`s and has an established functionality. What if there was a new rule `Ensures all ARIA role form elements are labeled`?

**Comment by WilcoFiers** [comment]
This needs to be a new rule. If we modify the existing rule to do this the remediation message for it won't be correct, it'll suggest adding implicit / explit label and the like. @straker can you use the rule template to write up a proposal?

https://github.com/dequelabs/axe-core/blob/develop/doc/rule-proposal.md

There are also some other "input related" roles to look at for this new rule. Another rule, very similar to this would be a best practice that tests non-widgets that the ARIA spec require an accessible name.

**Comment by WilcoFiers** [comment]
This rule should exclude elements that natively require an accessible name. We'll probably have to create a matches function that excludes things like input / select / textarea / button elements.

**Comment by dylanb** [comment]
Why not extend https://github.com/dequelabs/axe-core/pull/1610

**Comment by straker** [comment]
Closed by #1610
